data/domain/alerts.tsx

```diff
@@ -291,7 +292,8 @@ export class Alerts extends Domain {
     }
 }
 
-const getPlayerAlerts = (player: Player, anvil: AnvilWrapper, playerObols: Obol[], worshipData: Worship, prayers: Prayer[], divinityData: Divinity): Alert[] => {
+const getPlayerAlerts = (player: Player, anvil: AnvilWrapper, playerObols: Obol[], worshipData: Worship, prayers: Prayer[], divinityData: Divinity, companions: Companion[]): Alert[] => {
+    const dootOwned: boolean = companions.find(c => c.id === 0)?.owned ?? false;
```

Comment:
I don't want to tell you that your understanding of the code is wrong cause I don't believe it could be possible, but instead, I think you misunderstood what I wanted to achieve with those alerts.
To me, the checks need to combine the fact that Doot is owned or not and `gods.some` in some places.

Cause if Doot is owned, you don't have a choice on which god your char is linked to, while if you don't own it, then that means it's a player's choice and that it could be changed.

Take for example Bear god, it's only useful to be linked to it when you're not in the lab
Using `gods.some` will effectively let me know if the character is linked to this god in one way or another (directly linked, ES second god, owning Doot, or any other effect).
In the case where the player owns Doot, then he has no choice but to be linked to this god, so it won't serve any purpose to display the alert to him as he can't change anything about it (aside from deleting his Doot companion, but that'd be a really bad idea).
On the other hand, if he doesn't own it, then that means that he consciously chose to link to the Bear god instead of any other god that could be more useful for his progression, so displaying the alert becomes relevant.

Maybe I'm missing something with my reasoning, but to me, that's what the alert was created for
Ofc I value your point of view very highly as it's your site and I wouldn't dare to do something that you don't agree with, so tell me if you think it should be any different (don't hesitate to MP me on Discord, I think it would be better to talk using direct messages instead of comments)

Reply:
> I don't want to tell you that your understanding of the code is wrong cause I don't believe it could be possible

Totally possible haha, it's been a while since I wrote some of this code so I forget or even misremember my own intent so don't ever be afraid to call me out on it!

> I think you misunderstood what I wanted to achieve with those alerts.

It seems that was the case, your explanation makes sense to me and I now understand why you want to check for Doot directly in here. I'm fine with that reasoning so we can keep the alerts as is. Thanks for clarifying your intent in such detail!